runtime/src/iree/hal/drivers/cuda/tracing.h

```diff
@@ -62,80 +64,112 @@ void iree_hal_cuda_tracing_context_free(
 void iree_hal_cuda_tracing_context_collect(
     iree_hal_cuda_tracing_context_t* context);
 
+// Notify that the given command buffer has been submitted
+void iree_hal_cuda_tracing_notify_submitted(
+    iree_hal_cuda_tracing_context_t* context,
+    iree_hal_cuda_tracing_context_event_t** event_list_begin,
+    iree_hal_cuda_tracing_context_event_t** event_list_end);
```

Comment:
I see this pair repeating in many places. Wouldn't the the code become shorter/simpler if we used a struct `iree_hal_cuda_tracing_context_event_range_t` that encapsulates it. If you don't need to construct it in many but just pass it around it may help.